### PR TITLE
Enable HIP build in windows

### PR DESCRIPTION
**Pull request by rbramand-xilinx**
#### Problem solved by the commit
make changes to cmake files to enable HIP build on windows
Fix compilation errors

TODO: send submodule update with this changes to XRT-MCDM repo and enable hip build option in build-win.bat of MCDM repo

#### How problem was solved, alternative solutions (if any) and why they were rejected
In windows HIP SDK can be installed to get HIP include files and it install cmake config files of HIP to fixed path
For example latest SDK available on windows is of 5.7 version and it installs it to - C:/Program Files/AMD/ROCm/5.7/lib/cmake/hip
So when searching for HIP config files I have added this hardcode path but also gave an option for user to set environment variable HIP_DIR to pick config files from that location.

#### Risks (if any) associated the changes in the commit
Low

#### What has been tested and how, request additional testing if necessary
Tested xrt_flow test using HIP apis on phoenix using MCDM flow

#### Documentation impact (if any)
NA

### CODE REVIEW

File: src/CMake/nativeWin.cmake [hunk 1]
```diff
@@ -69,6 +69,18 @@ endif()
...
+if (XRT_ENABLE_HIP)
+  message("-- Looking for HIP include files...")
+  set(HIP_PLATFORM "amd")
+  # HIP SDK installs hip files to C:/Program Files/AMD/ROCm
+  # Latest version available on windows is 5.7
+  # Users can set HIP_DIR to location of HIP installation or default path is used
+  set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} $ENV{HIP_DIR} "C:/Program Files/AMD/ROCm/5.7/lib/cmake/hip")
+  include(hip-config)
+  message("-- Found at ${HIP_INCLUDE_DIR}")
+endif()
```
Comment (stsoe) [inline src/CMake/nativeWin.cmake:82]: I am no longer keen on distributed make snippets.  Is it possible to have this piece inside src/runtime_src/hip CMakeLists.txt or config.cmake instead?

File: src/runtime_src/hip/core/device.h [hunk 2]
```diff
@@ -3,6 +3,10 @@
 #ifndef xrthip_device_h
 #define xrthip_device_h
 
+#ifdef _WIN32
+#undef max
+#endif
```
Comment (stsoe) [inline src/runtime_src/hip/core/device.h:8]: This is annoying yes.   But I think the better fix is to #define NOMINMAX before inclusion of windows.h in all units that include this header.  I have seen this in Microsoft's own code:
```
#define NOMINMAX
#include <windows.h>
```

**Review by stsoe — CHANGES_REQUESTED** [review]
Few comments, take it or leave it.

File: src/runtime_src/hip/core/device.h (hunk 2 above)
Comment (rbramand-xilinx) [reply src/runtime_src/hip/core/device.h:8]: Hi @stsoe, when I remove this I get some different errors from MSVC internal code. So I haven't removed this. Hope that is okay

File: src/runtime_src/CMakeLists.txt [hunk 3]
```diff
@@ -40,7 +40,9 @@ if (XRT_DKMS_ALVEO STREQUAL "ON")
   xrt_add_subdirectory(ert)
 endif()
 
+# --- Optional HIP bindings ---
 if (XRT_ENABLE_HIP)
+  include(${CMAKE_CURRENT_SOURCE_DIR}/../CMake/hip.cmake)
```
Comment (stsoe) [inline src/runtime_src/CMakeLists.txt:45]: Can you move hip.cmake to src/runtime_src/hip/ some place and include it in src/runtime_src/hip/CMakeLists.txt?  We need to move away from global configuration.

File: src/runtime_src/CMakeLists.txt [hunk 4]
```diff
@@ -81,4 +83,10 @@ else()
   xrt_add_subdirectory(xdp)
   xrt_add_subdirectory(tools/scripts)
   xrt_add_subdirectory(core)
+
+  # --- Optional HIP bindings ---
+  if (XRT_ENABLE_HIP)
+    include(${CMAKE_CURRENT_SOURCE_DIR}/../CMake/hip.cmake)
```
Comment (stsoe) [inline src/runtime_src/CMakeLists.txt:89]: Ditto

**Review by stsoe — CHANGES_REQUESTED** [review]

File: src/runtime_src/CMakeLists.txt (hunk 3 above)
Comment (rbramand-xilinx) [reply src/runtime_src/CMakeLists.txt]: Done

**Review by stsoe — APPROVED** [review]

**Review by chvamshi-xilinx — APPROVED** [review]

**Pull request merged**